src/core/unit.c

```diff
@@ -5373,8 +5373,33 @@ void unit_remove_dependencies(Unit *u, UnitDependencyMask mask) {
         }
 }
 
+static int unit_get_invocation_path(Unit *u, char **path) {
+        char *p;
+
+        assert(u);
+        assert(path);
+
+        if (MANAGER_IS_SYSTEM(u->manager))
+                p = path_join("/run/systemd/units/invocation:", u->id);
+        else {
+                _cleanup_free_ char *user_path = NULL;
+                int r = xdg_user_runtime_dir(&user_path, "/systemd/units/invocation:");
```

Comment:
doesn't matter much, but usually we declare "int r" at the outermost scope, since it's a bit special, since it's our "default" variable for return values...